components/script/dom/bindings/frozenarray.rs

```diff
@@ -8,12 +8,13 @@ use js::jsval::JSVal;
 use js::rust::MutableHandleValue;
 
 use crate::dom::bindings::cell::DomRefCell;
+use crate::dom::bindings::trace::RootedTraceableBox;
 use crate::dom::bindings::utils::to_frozen_array;
 use crate::script_runtime::{CanGc, JSContext};
 
 #[derive(JSTraceable)]
 pub(crate) struct CachedFrozenArray {
```

Comment:
Here, and I think in many other places, you should not use a `RootedTraceableBox`, because the heap is immediately stored on a struct. What you should do instead is making sure the struct is `#[crown::unrooted_must_root_lint::must_root]`. I've checked this locally, and the previous code for this module compile, if you mark `CachedFrozenArray` as `must_root`.